### ISSUE TITLE
docker_volume resource does not work

### ISSUE THREAD

**Issue by EugenMayer** [chef-cookbooks/docker#786]
### Cookbook version
docker (2.13.0)

### Chef-client version
Chef: 12.16.42

### Platform Details
Debian 8.6 privisioned using test-kitchen

```yaml
driver:
  name: vagrant

provisioner:
  name: chef_zero
  # You may wish to disable always updating cookbooks in CI or other testing environments.
  # For example:
  #   always_update_cookbooks: <%= !ENV['CI'] %>
  always_update_cookbooks: true

verifier:
  name: inspec

platforms:
 debian-8.6 / x64
 Docker version 1.12.3, build 6b644ec
 default docker storage

```

### Scenario:
Iam trying to create docker volumes

### Steps to Reproduce:
When docker-engine is installed and the service is started (i can run docker_image and docker docker_registry without issues)

create a recipe like
```ruby
include_recipe 'chef-apt-docker'
include_recipe "docker"

package 'docker-engine' do
  action :install
end

service 'docker' do
  supports :status => true, :reload => true
  action [:enable, :start]
end

docker_volume 'dwdata' do
  action :create
end
```

### Expected Result:
A docker-volume should be created

### Actual Result:

```
 ================================================================================
           Error executing action `create` on resource 'docker_volume[dwdata]'
           ================================================================================
           
           NoMethodError
           -------------
           undefined method `[]=' for #<Docker::Connection:0x00000004139d78>
           
           Cookbook Trace:
           ---------------
           /tmp/kitchen/cache/cookbooks/docker/libraries/docker_volume.rb:23:in `block (2 levels) in <class:DockerVolume>'
           /tmp/kitchen/cache/cookbooks/docker/libraries/docker_volume.rb:22:in `block in <class:DockerVolume>'
           
           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/dw-host/recipes/docker_volumes.rb
           
             1: docker_volume 'dwdata' do
             2:   action :create
             3: end
             4: 
           
           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/dw-host/recipes/docker_volumes.rb:1:in `from_file'
           
           docker_volume("dwdata") do
             action [:create]
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             declared_type :docker_volume
             cookbook_name "dw-host"
             recipe_name "docker_volumes"
             volume_name "dwdata"
             connection #<Docker::Connection:0x00000004139d78 @url="unix:///", @options={:socket=>"/var/run/docker.sock", :read_timeout=>60}>
           end
           
           Platform:
           ---------
           x86_64-linux

```

**Comment by s-bernard** [comment]
Hi, this issue comes from "Updating docker-api gem dep 1.32.1" in 2.12.0. Reverting to 2.11.1 fixes it.

**Comment by someara** [comment]
Fixed up in bea482de8b164c4ffc567d47b0d92e62d1d4d18d

**Comment by someara** [comment]
Releasing in 2.13.1
